zcash_client_backend/src/keys.rs

```diff
@@ -137,38 +140,135 @@ impl UnifiedSpendingKey {
 #[derive(Clone, Debug)]
 #[doc(hidden)]
 pub struct UnifiedFullViewingKey {
-    account: AccountId,
     #[cfg(feature = "transparent-inputs")]
     transparent: Option<legacy::AccountPubKey>,
-    // TODO: This type is invalid for a UFVK; create a `sapling::DiversifiableFullViewingKey`
-    // to replace it.
-    sapling: Option<sapling::ExtendedFullViewingKey>,
+    sapling: Option<sapling_keys::DiversifiableFullViewingKey>,
+    orchard: Option<orchard::keys::FullViewingKey>,
+    unknown: Vec<(u32, Vec<u8>)>,
```

Comment:
I still don't think we should include the unknown components here, nor should we seek roundtrip serialization capability at this level.

The data flow I envision for UFVK import (the only case where this might matter) is that the original encoding that was imported is persisted, and then it is decoded to a type which does not contain the unknown parts. That decoded value should be in-memory only, and should never need to be reencoded; if the encoded form is required, the original encoded value should be used.